axelor-supplychain/src/main/java/com/axelor/apps/supplychain/web/MrpLineController.java

```diff
@@ -38,4 +48,72 @@ public void generateProposal(ActionRequest request, ActionResponse response)
     response.setFlash(I18n.get("The proposal has been successfully generated."));
     response.setReload(true);
   }
+
+  @Transactional(rollbackOn = {Exception.class})
+  public void select(ActionRequest request, ActionResponse response) {
+    toggle(request, response, true);
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  public void unselect(ActionRequest request, ActionResponse response) {
+    toggle(request, response, false);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Transactional(rollbackOn = {Exception.class})
+  private void toggle(ActionRequest request, ActionResponse response, boolean value) {
+    List<Integer> mrpLineIds = (List<Integer>) request.getContext().get("_ids");
+
+    if (CollectionUtils.isNotEmpty(mrpLineIds)) {
+      MrpLineRepository mrpLineRepo = Beans.get(MrpLineRepository.class);
+      MrpLine mrpLine;
+
+      for (Integer mrpId : mrpLineIds) {
+        mrpLine = mrpLineRepo.find(Long.valueOf(mrpId));
+        mrpLine.setToProcess(value);
+        mrpLineRepo.save(mrpLine);
+      }
+    }
+
+    response.setAttr("mrpLinePanel", "refresh", true);
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  public void selectAll(ActionRequest request, ActionResponse response) {
+    try {
+      MrpLineRepository mrpLineRepo = Beans.get(MrpLineRepository.class);
+      Mrp mrp = request.getContext().getParent().asType(Mrp.class);
+      mrp = Beans.get(MrpRepository.class).find(mrp.getId());
+
+      Query<MrpLine> mrpLineQuery =
+          mrpLineRepo
+              .all()
+              .filter(
+                  "(self.mrp.displayProductWithoutProposal = true AND self.mrp.id = ?1)"
+                      + " OR (self.mrp.displayProductWithoutProposal = false AND self.mrp.id = ?1 AND self.product.id IN (select m.product from MrpLine as m where m.mrp.id = ?1 AND m.mrpLineType.elementSelect in (?2, ?3, ?4)))",
```

Comment:
this line is too wide, please split it to follow our style convention.
Also please use named parameters to improve maintainability and readability.